src/text-filter/search-results.tsx

```diff
@@ -15,12 +15,52 @@ interface SearchResultsProps {
   children: string;
 }
 
-export function SearchResults({ id, children }: SearchResultsProps) {
-  return (
-    <span className={styles.results}>
-      <InternalLiveRegion delay={LIVE_REGION_DELAY} tagName="span">
-        <span id={id}>{children}</span>
-      </InternalLiveRegion>
-    </span>
-  );
-}
+export const SearchResults = React.forwardRef(
+  ({ id, children }: SearchResultsProps, ref?: React.Ref<InternalLiveRegionRef>) => {
+    const [announcementText, setAnnouncementText] = useState('');
+    const isComponentMounted = useRef(false);
+    const liveRegionRef = useRef<InternalLiveRegionRef>(null);
+
+    useEffect(() => {
+      isComponentMounted.current = true;
```

Comment:
It could happen, the component is unmounted and a debounce is pending which could cause some console warnings. To prevent, I keep track of the component is mounted or not.